code/modules/pda/mob_hunt_game_app.dm

```diff
@@ -138,12 +144,27 @@
 	card.forceMove(get_turf(pda))
 	remove_mob()
 
-/datum/data/pda/app/mob_hunter_game/proc/remove_mob()
-	if(!my_collection.len)
+/**
+  * Removes a Nanomob from the [my_collection] list.
+  *
+  * The Nanomob that is currently selected in the app ([current_index]) will be removed from the list unless a `mob_override` argument is given, in which case that will be removed instead.
+  *
+  * Arguments:
+  * * mob_override - A specific Nanomob to remove from the list. (Optional)
+  */
+/datum/data/pda/app/mob_hunter_game/proc/remove_mob(datum/mob_hunt/mob_override = null)
+	SIGNAL_HANDLER
+	var/collection_length = length(my_collection)
+	if(!collection_length)
 		return
-	my_collection.Remove(my_collection[current_index])
-	if(current_index > my_collection.len)
-		current_index = my_collection.len
+
+	if(mob_override)
+		my_collection -= mob_override
+	else
+		my_collection -= my_collection[current_index]
+
+	if(current_index > collection_length)
+		current_index = collection_length
```

Comment:
I made a mistake here:
The length of the `my_collection` list only changes *after* the `collection_length` variable is defined, so the check on line 166 never passes.
I actually had this working properly originally, but I reverted it before pushing since it seemed to make more sense. Downside of leaving WIP code for half a week with no notes I guess.

I'll fix this when I get up tomorrow since I can't do it from my phone. Should just be able to move the `collection_length` variable to line 165.